stdlib/http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/WebSocketConnector.java

```diff
@@ -127,8 +127,25 @@ public static Object pong(ObjectValue wsConnection, ArrayValue binaryData) {
                                                     WebSocketObservabilityConstants.ERROR_TYPE_MESSAGE_SENT,
                                                     WebSocketObservabilityConstants.MESSAGE_TYPE_PONG,
                                                     e.getMessage());
-            callback.notifyFailure(WebSocketUtil.createErrorByType(e));
+            setCallback(connectionInfo, callback, e);
         }
         return null;
     }
+
+    public static void setCallback(WebSocketConnectionInfo connectionInfo, NonBlockingCallback callback,
```

Comment:
Between if this going to be a utility function used by other classes we can move it to the WebSocketUtil class.